### DISCUSSION TITLE
useInfiniteQuery Modify Custom Props Not Work Correctly

### DISCUSSION THREAD

**Discussion by dim0147** [TanStack/query#2303] [Q&A]
**Describe the bug**
- Not sure it's a bug or I missing knowledge but when using **useInfiniteQuery** at second parameters 
![image](https://user-images.githubusercontent.com/44487221/119088518-624d4500-ba3b-11eb-906a-76bc32629c82.png)
here is **fetchMovies** when I assign new properties from custom prop like this 
![image](https://user-images.githubusercontent.com/44487221/119088708-aa6c6780-ba3b-11eb-8682-6493d89fa2a2.png)
the fetch won't fetch correctly, it keep fetching the initialize result. to fix this have to **clone** the object
![image](https://user-images.githubusercontent.com/44487221/119088823-db4c9c80-ba3b-11eb-8134-e6db24d42f0b.png)

**Comment by dim0147** [comment]
Note: the **querySearch** is the prop of component, I pass this prop into **useInfiniteQuery** 
![image](https://user-images.githubusercontent.com/44487221/119089073-31b9db00-ba3c-11eb-8eed-dee2fa3e79da.png)

**Comment by TkDodo** [comment]
sorry, I have no idea what you are trying to achieve. converting this to a discussion so that we can discuss it.

`pageParam` is "calculated" in the `getNextPageParam` function - have you implemented that? The result of that function is passed as `pageParam` to the `queryFn` ....

**Reply by dim0147** [reply]
I have implement `getNextPageParam`
![image](https://user-images.githubusercontent.com/44487221/119090455-3bdcd900-ba3e-11eb-8796-7e8405dfa8ff.png)
 and I think it not related to this problem, here you can see I receive `querySearch` props from the parent component and pass it into `useInfiniteQuery` as **query-key** (second index of item array), in `fetchMovies` I receive the `querySearch` object as `queryKey[1]`, I want to add one more custom property into `queryKey[1]` object like the image above and the fetch not working correctly (always fetch the first page) so I using **spread** operator to **speard** the `queryKey[1]` into a new variable and add new property into that new variable the fetch working correctly.

**Reply by TkDodo** [reply]
> I want to add one more custom property into queryKey[1] object

Very hard for me to understand that. It seems like you are attempting this as a solution to a problem, but I don't know what problem you are trying to solve, so it's hard to help. Also, more screenshots are not helping. Can you provide a description of the problem you are having, possibly with a minimal running codesandbox example that shows the issue?

**Reply by dim0147** [reply]
- I make a problem, you can check 
[https://codesandbox.io/s/pensive-mountain-mj9p3?file=/src/App.js](https://codesandbox.io/s/pensive-mountain-mj9p3?file=/src/App.js)
- The two cases :
![image](https://user-images.githubusercontent.com/44487221/119207346-7db65f80-bad0-11eb-9529-148b248abd3f.png)

- Note the red one, uncomment the first one  and it fetch fine, the second is not working

**Reply by TkDodo** [reply]
What I’m seeing is that in line 17, you mutate objQuery:

`objQuery["_start"] = start;`

If objQuery is a reference to queryKey[1], it will also mutate the queryKey. Stick to immutable programming :)
